File: schemaorg-rs/src/patient.rs
use crate::*;
use serde_with::{serde_as, OneOrMany};
///<https://schema.org/additionalType>
///<https://schema.org/Text>
///<https://schema.org/URL>
pub type PatientAdditionalTypeFieldEnum = String;
///<https://schema.org/Patient>
#[serde_as]
#[derive(Debug, serde::Deserialize)]
#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
pub struct Patient {
    #[serde(rename = "@context")]
    pub context: String,
    ///<https://schema.org/workLocation>
    #[serde(rename = "workLocation")]
    #[serde_as(as = "OneOrMany<_>")]
    pub work_location: Vec<PatientWorkLocationFieldEnum>,
    ///<https://schema.org/colleagues>
    #[serde(rename = "colleagues")]
    #[serde_as(as = "OneOrMany<_>")]
    pub colleagues: Vec<Person>,
    ///<https://schema.org/spouse>
    #[serde(rename = "spouse")]
    #[serde_as(as = "OneOrMany<_>")]
    pub spouse: Vec<Person>,
    ///<https://schema.org/honorificSuffix>
    #[serde(rename = "honorificSuffix")]
    #[serde_as(as = "OneOrMany<_>")]
    pub honorific_suffix: Vec<String>,
    ///<https://schema.org/drug>
    #[serde(rename = "drug")]
    #[serde_as(as = "OneOrMany<_>")]
    pub drug: Vec<Drug>,
    ///<https://schema.org/parents>
    #[serde(rename = "parents")]
    #[serde_as(as = "OneOrMany<_>")]
    pub parents: Vec<Person>,
    ///<https://schema.org/hasOccupation>
    #[serde(rename = "hasOccupation")]
    #[serde_as(as = "OneOrMany<_>")]
    pub has_occupation: Vec<Occupation>,
    ///<https://schema.org/seeks>
    #[serde(rename = "seeks")]
    #[serde_as(as = "OneOrMany<_>")]
    pub seeks: Vec<Demand>,
    ///<https://schema.org/callSign>
    #[serde(rename = "callSign")]
    #[serde_as(as = "OneOrMany<_>")]
    pub call_sign: Vec<String>,
    ///<https://schema.org/interactionStatistic>
    #[serde(rename = "interactionStatistic")]
    #[serde_as(as = "OneOrMany<_>")]
    pub interaction_statistic: Vec<InteractionCounter>,
    ///<https://schema.org/deathPlace>
    #[serde(rename = "deathPlace")]
    #[serde_as(as = "OneOrMany<_>")]
    pub death_place: Vec<Place>,
    ///<https://schema.org/gender>
    #[serde(rename = "gender")]
    #[serde_as(as = "OneOrMany<_>")]
    pub gender: Vec<PatientGenderFieldEnum>,
    ///<https://schema.org/givenName>
    #[serde(rename = "givenName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub given_name: Vec<String>,
    ///<https://schema.org/netWorth>
    #[serde(rename = "netWorth")]
    #[serde_as(as = "OneOrMany<_>")]
    pub net_worth: Vec<PatientNetWorthFieldEnum>,
    ///<https://schema.org/name>
    #[serde(rename = "name")]
    #[serde_as(as = "OneOrMany<_>")]
    pub name: Vec<String>,
    ///<https://schema.org/honorificPrefix>
    #[serde(rename = "honorificPrefix")]
    #[serde_as(as = "OneOrMany<_>")]
    pub honorific_prefix: Vec<String>,
    ///<https://schema.org/disambiguatingDescription>
    #[serde(rename = "disambiguatingDescription")]
    #[serde_as(as = "OneOrMany<_>")]
    pub disambiguating_description: Vec<String>,
    ///<https://schema.org/sibling>
    #[serde(rename = "sibling")]
    #[serde_as(as = "OneOrMany<_>")]
    pub sibling: Vec<Person>,
    ///<https://schema.org/homeLocation>
    #[serde(rename = "homeLocation")]
    #[serde_as(as = "OneOrMany<_>")]
    pub home_location: Vec<PatientHomeLocationFieldEnum>,
    ///<https://schema.org/potentialAction>
    #[serde(rename = "potentialAction")]
    #[serde_as(as = "OneOrMany<_>")]
    pub potential_action: Vec<Action>,
    ///<https://schema.org/requiredMinAge>
    #[serde(rename = "requiredMinAge")]
    #[serde_as(as = "OneOrMany<_>")]
    pub required_min_age: Vec<i32>,
    ///<https://schema.org/telephone>
    #[serde(rename = "telephone")]
    #[serde_as(as = "OneOrMany<_>")]
    pub telephone: Vec<String>,
    ///<https://schema.org/colleague>
    #[serde(rename = "colleague")]
    #[serde_as(as = "OneOrMany<_>")]
    pub colleague: Vec<PatientColleagueFieldEnum>,
    ///<https://schema.org/description>
    #[serde(rename = "description")]
    #[serde_as(as = "OneOrMany<_>")]
    pub description: Vec<PatientDescriptionFieldEnum>,
    ///<https://schema.org/geographicArea>
    #[serde(rename = "geographicArea")]
    #[serde_as(as = "OneOrMany<_>")]
    pub geographic_area: Vec<AdministrativeArea>,
    ///<https://schema.org/performerIn>
    #[serde(rename = "performerIn")]
    #[serde_as(as = "OneOrMany<_>")]
    pub performer_in: Vec<Event>,
    ///<https://schema.org/isicV4>
    #[serde(rename = "isicV4")]
    #[serde_as(as = "OneOrMany<_>")]
    pub isic_v4: Vec<String>,
    ///<https://schema.org/hasCertification>
    #[serde(rename = "hasCertification")]
    #[serde_as(as = "OneOrMany<_>")]
    pub has_certification: Vec<Certification>,
    ///<https://schema.org/awards>
    #[serde(rename = "awards")]
    #[serde_as(as = "OneOrMany<_>")]
    pub awards: Vec<String>,
    ///<https://schema.org/deathDate>
    #[serde(rename = "deathDate")]
    #[serde_as(as = "OneOrMany<_>")]
    pub death_date: Vec<String>,
    ///<https://schema.org/subjectOf>
    #[serde(rename = "subjectOf")]
    #[serde_as(as = "OneOrMany<_>")]
    pub subject_of: Vec<PatientSubjectOfFieldEnum>,
    ///<https://schema.org/mainEntityOfPage>
    #[serde(rename = "mainEntityOfPage")]
    #[serde_as(as = "OneOrMany<_>")]
    pub main_entity_of_page: Vec<PatientMainEntityOfPageFieldEnum>,
    ///<https://schema.org/weight>
    #[serde(rename = "weight")]
    #[serde_as(as = "OneOrMany<_>")]
    pub weight: Vec<PatientWeightFieldEnum>,
    ///<https://schema.org/memberOf>
    #[serde(rename = "memberOf")]
    #[serde_as(as = "OneOrMany<_>")]
    pub member_of: Vec<PatientMemberOfFieldEnum>,
    ///<https://schema.org/jobTitle>
    #[serde(rename = "jobTitle")]
    #[serde_as(as = "OneOrMany<_>")]
    pub job_title: Vec<PatientJobTitleFieldEnum>,
    ///<https://schema.org/additionalName>
    #[serde(rename = "additionalName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub additional_name: Vec<String>,
    ///<https://schema.org/globalLocationNumber>
    #[serde(rename = "globalLocationNumber")]
    #[serde_as(as = "OneOrMany<_>")]
    pub global_location_number: Vec<String>,
    ///<https://schema.org/hasPOS>
    #[serde(rename = "hasPOS")]
    #[serde_as(as = "OneOrMany<_>")]
    pub has_pos: Vec<Place>,
    ///<https://schema.org/sameAs>
    #[serde(rename = "sameAs")]
    #[serde_as(as = "OneOrMany<_>")]
    pub same_as: Vec<String>,
    ///<https://schema.org/requiredMaxAge>
    #[serde(rename = "requiredMaxAge")]
    #[serde_as(as = "OneOrMany<_>")]
    pub required_max_age: Vec<i32>,
    ///<https://schema.org/url>
    #[serde(rename = "url")]
    #[serde_as(as = "OneOrMany<_>")]
    pub url: Vec<String>,
    ///<https://schema.org/suggestedMeasurement>
    #[serde(rename = "suggestedMeasurement")]
    #[serde_as(as = "OneOrMany<_>")]
    pub suggested_measurement: Vec<QuantitativeValue>,
    ///<https://schema.org/parent>
    #[serde(rename = "parent")]
    #[serde_as(as = "OneOrMany<_>")]
    pub parent: Vec<Person>,
    ///<https://schema.org/agentInteractionStatistic>
    #[serde(rename = "agentInteractionStatistic")]
    #[serde_as(as = "OneOrMany<_>")]
    pub agent_interaction_statistic: Vec<InteractionCounter>,
    ///<https://schema.org/image>
    #[serde(rename = "image")]
    #[serde_as(as = "OneOrMany<_>")]
    pub image: Vec<PatientImageFieldEnum>,
    ///<https://schema.org/funder>
    #[serde(rename = "funder")]
    #[serde_as(as = "OneOrMany<_>")]
    pub funder: Vec<PatientFunderFieldEnum>,
    ///<https://schema.org/birthDate>
    #[serde(rename = "birthDate")]
    #[serde_as(as = "OneOrMany<_>")]
    pub birth_date: Vec<String>,
    ///<https://schema.org/suggestedAge>
    #[serde(rename = "suggestedAge")]
    #[serde_as(as = "OneOrMany<_>")]
    pub suggested_age: Vec<QuantitativeValue>,
    ///<https://schema.org/funding>
    #[serde(rename = "funding")]
    #[serde_as(as = "OneOrMany<_>")]
    pub funding: Vec<Grant>,
    ///<https://schema.org/suggestedGender>
    #[serde(rename = "suggestedGender")]
    #[serde_as(as = "OneOrMany<_>")]
    pub suggested_gender: Vec<PatientSuggestedGenderFieldEnum>,
    ///<https://schema.org/sponsor>
    #[serde(rename = "sponsor")]
    #[serde_as(as = "OneOrMany<_>")]
    pub sponsor: Vec<PatientSponsorFieldEnum>,
    ///<https://schema.org/affiliation>
    #[serde(rename = "affiliation")]
    #[serde_as(as = "OneOrMany<_>")]
    pub affiliation: Vec<Organization>,
    ///<https://schema.org/children>
    #[serde(rename = "children")]
    #[serde_as(as = "OneOrMany<_>")]
    pub children: Vec<Person>,
    ///<https://schema.org/familyName>
    #[serde(rename = "familyName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub family_name: Vec<String>,
    ///<https://schema.org/hasOfferCatalog>
    #[serde(rename = "hasOfferCatalog")]
    #[serde_as(as = "OneOrMany<_>")]
    pub has_offer_catalog: Vec<OfferCatalog>,
    ///<https://schema.org/contactPoints>
    #[serde(rename = "contactPoints")]
    #[serde_as(as = "OneOrMany<_>")]
    pub contact_points: Vec<ContactPoint>,
    ///<https://schema.org/owns>
    #[serde(rename = "owns")]
    #[serde_as(as = "OneOrMany<_>")]
    pub owns: Vec<PatientOwnsFieldEnum>,
    ///<https://schema.org/faxNumber>
    #[serde(rename = "faxNumber")]
    #[serde_as(as = "OneOrMany<_>")]
    pub fax_number: Vec<String>,
    ///<https://schema.org/nationality>
    #[serde(rename = "nationality")]
    #[serde_as(as = "OneOrMany<_>")]
    pub nationality: Vec<Country>,
    ///<https://schema.org/knowsLanguage>
    #[serde(rename = "knowsLanguage")]
    #[serde_as(as = "OneOrMany<_>")]
    pub knows_language: Vec<PatientKnowsLanguageFieldEnum>,
    ///<https://schema.org/follows>
    #[serde(rename = "follows")]
    #[serde_as(as = "OneOrMany<_>")]
    pub follows: Vec<Person>,
    ///<https://schema.org/award>
    #[serde(rename = "award")]
    #[serde_as(as = "OneOrMany<_>")]
    pub award: Vec<String>,
    ///<https://schema.org/taxID>
    #[serde(rename = "taxID")]
    #[serde_as(as = "OneOrMany<_>")]
    pub tax_id: Vec<String>,
    ///<https://schema.org/alternateName>
    #[serde(rename = "alternateName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub alternate_name: Vec<String>,
    ///<https://schema.org/additionalType>
    #[serde(rename = "additionalType")]
    #[serde_as(as = "OneOrMany<_>")]
    pub additional_type: Vec<PatientAdditionalTypeFieldEnum>,
    ///<https://schema.org/suggestedMaxAge>
    #[serde(rename = "suggestedMaxAge")]
    #[serde_as(as = "OneOrMany<_>")]
    pub suggested_max_age: Vec<f32>,
    ///<https://schema.org/diagnosis>
    #[serde(rename = "diagnosis")]
    #[serde_as(as = "OneOrMany<_>")]
    pub diagnosis: Vec<MedicalCondition>,
    ///<https://schema.org/siblings>
    #[serde(rename = "siblings")]
    #[serde_as(as = "OneOrMany<_>")]
    pub siblings: Vec<Person>,
    ///<https://schema.org/height>
    #[serde(rename = "height")]
    #[serde_as(as = "OneOrMany<_>")]
    pub height: Vec<PatientHeightFieldEnum>,
    ///<https://schema.org/alumniOf>
    #[serde(rename = "alumniOf")]
    #[serde_as(as = "OneOrMany<_>")]
    pub alumni_of: Vec<PatientAlumniOfFieldEnum>,
    ///<https://schema.org/makesOffer>
    #[serde(rename = "makesOffer")]
    #[serde_as(as = "OneOrMany<_>")]
    pub makes_offer: Vec<Offer>,
    ///<https://schema.org/knows>
    #[serde(rename = "knows")]
    #[serde_as(as = "OneOrMany<_>")]
    pub knows: Vec<Person>,
    ///<https://schema.org/skills>
    #[serde(rename = "skills")]
    #[serde_as(as = "OneOrMany<_>")]
    pub skills: Vec<PatientSkillsFieldEnum>,
    ///<https://schema.org/contactPoint>
    #[serde(rename = "contactPoint")]
    #[serde_as(as = "OneOrMany<_>")]
    pub contact_point: Vec<ContactPoint>,
    ///<https://schema.org/email>
    #[serde(rename = "email")]
    #[serde_as(as = "OneOrMany<_>")]
    pub email: Vec<String>,
    ///<https://schema.org/publishingPrinciples>
    #[serde(rename = "publishingPrinciples")]
    #[serde_as(as = "OneOrMany<_>")]
    pub publishing_principles: Vec<PatientPublishingPrinciplesFieldEnum>,
    ///<https://schema.org/vatID>
    #[serde(rename = "vatID")]
    #[serde_as(as = "OneOrMany<_>")]
    pub vat_id: Vec<String>,
    ///<https://schema.org/knowsAbout>
    #[serde(rename = "knowsAbout")]
    #[serde_as(as = "OneOrMany<_>")]
    pub knows_about: Vec<PatientKnowsAboutFieldEnum>,
    ///<https://schema.org/relatedTo>
    #[serde(rename = "relatedTo")]
    #[serde_as(as = "OneOrMany<_>")]
    pub related_to: Vec<Person>,
    ///<https://schema.org/suggestedMinAge>
    #[serde(rename = "suggestedMinAge")]
    #[serde_as(as = "OneOrMany<_>")]
    pub suggested_min_age: Vec<f32>,
    ///<https://schema.org/naics>
    #[serde(rename = "naics")]
    #[serde_as(as = "OneOrMany<_>")]
    pub naics: Vec<String>,
    ///<https://schema.org/worksFor>
    #[serde(rename = "worksFor")]
    #[serde_as(as = "OneOrMany<_>")]
    pub works_for: Vec<Organization>,
    ///<https://schema.org/brand>
    #[serde(rename = "brand")]
    #[serde_as(as = "OneOrMany<_>")]
    pub brand: Vec<PatientBrandFieldEnum>,
    ///<https://schema.org/hasCredential>
    #[serde(rename = "hasCredential")]
    #[serde_as(as = "OneOrMany<_>")]
    pub has_credential: Vec<EducationalOccupationalCredential>,
    ///<https://schema.org/address>
    #[serde(rename = "address")]
    #[serde_as(as = "OneOrMany<_>")]
    pub address: Vec<PatientAddressFieldEnum>,
    ///<https://schema.org/duns>
    #[serde(rename = "duns")]
    #[serde_as(as = "OneOrMany<_>")]
    pub duns: Vec<String>,
    ///<https://schema.org/identifier>
    #[serde(rename = "identifier")]
    #[serde_as(as = "OneOrMany<_>")]
    pub identifier: Vec<PatientIdentifierFieldEnum>,
    ///<https://schema.org/audienceType>
    #[serde(rename = "audienceType")]
    #[serde_as(as = "OneOrMany<_>")]
    pub audience_type: Vec<String>,
    ///<https://schema.org/birthPlace>
    #[serde(rename = "birthPlace")]
    #[serde_as(as = "OneOrMany<_>")]
    pub birth_place: Vec<Place>,
    ///<https://schema.org/healthCondition>
    #[serde(rename = "healthCondition")]
    #[serde_as(as = "OneOrMany<_>")]
    pub health_condition: Vec<MedicalCondition>,
    ///<https://schema.org/requiredGender>
    #[serde(rename = "requiredGender")]
    #[serde_as(as = "OneOrMany<_>")]
    pub required_gender: Vec<String>,
}

File: schemaorg-ts/src/patient.ts
import type * as all from "./index";

/**
 * <https://schema.org/additionalType>
 **/
export type PatientAdditionalTypeFieldEnum = string;

/**
 * <https://schema.org/Patient>
 **/
export interface Patient {
  "@context": string;
  "@type": "Patient";

  /**
   * <https://schema.org/healthCondition>
   **/
  healthCondition: all.OneOrMany<all.MedicalCondition>;

  /**
   * <https://schema.org/drug>
   **/
  drug: all.OneOrMany<all.Drug>;

  /**
   * <https://schema.org/diagnosis>
   **/
  diagnosis: all.OneOrMany<all.MedicalCondition>;

  /**
   * <https://schema.org/sibling>
   **/
  sibling: all.OneOrMany<all.Person>;

  /**
   * <https://schema.org/knows>
   **/
  knows: all.OneOrMany<all.Person>;

  /**
   * <https://schema.org/isicV4>
   **/
  isicV4: all.OneOrMany<string>;

  /**
   * <https://schema.org/nationality>
   **/
  nationality: all.OneOrMany<all.Country>;

  /**
   * <https://schema.org/honorificPrefix>
   **/
  honorificPrefix: all.OneOrMany<string>;

  /**
   * <https://schema.org/siblings>
   **/
  siblings: all.OneOrMany<all.Person>;

  /**
   * <https://schema.org/colleague>
   **/
  colleague: all.OneOrMany<all.PatientColleagueFieldEnum>;

  /**
   * <https://schema.org/hasCertification>
   **/
  hasCertification: all.OneOrMany<all.Certification>;

  /**
   * <https://schema.org/parent>
   **/
  parent: all.OneOrMany<all.Person>;

  /**
   * <https://schema.org/skills>
   **/
  skills: all.OneOrMany<all.PatientSkillsFieldEnum>;

  /**
   * <https://schema.org/publishingPrinciples>
   **/
  publishingPrinciples: all.OneOrMany<all.PatientPublishingPrinciplesFieldEnum>;

  /**
   * <https://schema.org/birthPlace>
   **/
  birthPlace: all.OneOrMany<all.Place>;

  /**
   * <https://schema.org/hasPOS>
   **/
  hasPOS: all.OneOrMany<all.Place>;

  /**
   * <https://schema.org/knowsAbout>
   **/
  knowsAbout: all.OneOrMany<all.PatientKnowsAboutFieldEnum>;

  /**
   * <https://schema.org/telephone>
   **/
  telephone: all.OneOrMany<string>;

  /**
   * <https://schema.org/weight>
   **/
  weight: all.OneOrMany<all.PatientWeightFieldEnum>;

  /**
   * <https://schema.org/seeks>
   **/
  seeks: all.OneOrMany<all.Demand>;

  /**
   * <https://schema.org/additionalName>
   **/
  additionalName: all.OneOrMany<string>;

  /**
   * <https://schema.org/hasCredential>
   **/
  hasCredential: all.OneOrMany<all.EducationalOccupationalCredential>;

  /**
   * <https://schema.org/deathDate>
   **/
  deathDate: all.OneOrMany<string>;

  /**
   * <https://schema.org/jobTitle>
   **/
  jobTitle: all.OneOrMany<all.PatientJobTitleFieldEnum>;

  /**
   * <https://schema.org/vatID>
   **/
  vatID: all.OneOrMany<string>;

  /**
   * <https://schema.org/taxID>
   **/
  taxID: all.OneOrMany<string>;

  /**
   * <https://schema.org/colleagues>
   **/
  colleagues: all.OneOrMany<all.Person>;

  /**
   * <https://schema.org/funding>
   **/
  funding: all.OneOrMany<all.Grant>;

  /**
   * <https://schema.org/givenName>
   **/
  givenName: all.OneOrMany<string>;

  /**
   * <https://schema.org/relatedTo>
   **/
  relatedTo: all.OneOrMany<all.Person>;

  /**
   * <https://schema.org/birthDate>
   **/
  birthDate: all.OneOrMany<string>;

  /**
   * <https://schema.org/homeLocation>
   **/
  homeLocation: all.OneOrMany<all.PatientHomeLocationFieldEnum>;

  /**
   * <https://schema.org/alumniOf>
   **/
  alumniOf: all.OneOrMany<all.PatientAlumniOfFieldEnum>;

  /**
   * <https://schema.org/contactPoints>
   **/
  contactPoints: all.OneOrMany<all.ContactPoint>;

  /**
   * <https://schema.org/naics>
   **/
  naics: all.OneOrMany<string>;

  /**
   * <https://schema.org/owns>
   **/
  owns: all.OneOrMany<all.PatientOwnsFieldEnum>;

  /**
   * <https://schema.org/callSign>
   **/
  callSign: all.OneOrMany<string>;

  /**
   * <https://schema.org/award>
   **/
  award: all.OneOrMany<string>;

  /**
   * <https://schema.org/brand>
   **/
  brand: all.OneOrMany<all.PatientBrandFieldEnum>;

  /**
   * <https://schema.org/children>
   **/
  children: all.OneOrMany<all.Person>;

  /**
   * <https://schema.org/knowsLanguage>
   **/
  knowsLanguage: all.OneOrMany<all.PatientKnowsLanguageFieldEnum>;

  /**
   * <https://schema.org/awards>
   **/
  awards: all.OneOrMany<string>;

  /**
   * <https://schema.org/affiliation>
   **/
  affiliation: all.OneOrMany<all.Organization>;

  /**
   * <https://schema.org/parents>
   **/
  parents: all.OneOrMany<all.Person>;

  /**
   * <https://schema.org/spouse>
   **/
  spouse: all.OneOrMany<all.Person>;

  /**
   * <https://schema.org/agentInteractionStatistic>
   **/
  agentInteractionStatistic: all.OneOrMany<all.InteractionCounter>;

  /**
   * <https://schema.org/funder>
   **/
  funder: all.OneOrMany<all.PatientFunderFieldEnum>;

  /**
   * <https://schema.org/address>
   **/
  address: all.OneOrMany<all.PatientAddressFieldEnum>;

  /**
   * <https://schema.org/netWorth>
   **/
  netWorth: all.OneOrMany<all.PatientNetWorthFieldEnum>;

  /**
   * <https://schema.org/interactionStatistic>
   **/
  interactionStatistic: all.OneOrMany<all.InteractionCounter>;

  /**
   * <https://schema.org/globalLocationNumber>
   **/
  globalLocationNumber: all.OneOrMany<string>;

  /**
   * <https://schema.org/hasOccupation>
   **/
  hasOccupation: all.OneOrMany<all.Occupation>;

  /**
   * <https://schema.org/deathPlace>
   **/
  deathPlace: all.OneOrMany<all.Place>;

  /**
   * <https://schema.org/makesOffer>
   **/
  makesOffer: all.OneOrMany<all.Offer>;

  /**
   * <https://schema.org/sponsor>
   **/
  sponsor: all.OneOrMany<all.PatientSponsorFieldEnum>;

  /**
   * <https://schema.org/memberOf>
   **/
  memberOf: all.OneOrMany<all.PatientMemberOfFieldEnum>;

  /**
   * <https://schema.org/hasOfferCatalog>
   **/
  hasOfferCatalog: all.OneOrMany<all.OfferCatalog>;

  /**
   * <https://schema.org/height>
   **/
  height: all.OneOrMany<all.PatientHeightFieldEnum>;

  /**
   * <https://schema.org/duns>
   **/
  duns: all.OneOrMany<string>;

  /**
   * <https://schema.org/familyName>
   **/
  familyName: all.OneOrMany<string>;

  /**
   * <https://schema.org/worksFor>
   **/
  worksFor: all.OneOrMany<all.Organization>;

  /**
   * <https://schema.org/follows>
   **/
  follows: all.OneOrMany<all.Person>;

  /**
   * <https://schema.org/faxNumber>
   **/
  faxNumber: all.OneOrMany<string>;

  /**
   * <https://schema.org/contactPoint>
   **/
  contactPoint: all.OneOrMany<all.ContactPoint>;

  /**
   * <https://schema.org/honorificSuffix>
   **/
  honorificSuffix: all.OneOrMany<string>;

  /**
   * <https://schema.org/performerIn>
   **/
  performerIn: all.OneOrMany<all.Event>;

  /**
   * <https://schema.org/email>
   **/
  email: all.OneOrMany<string>;

  /**
   * <https://schema.org/workLocation>
   **/
  workLocation: all.OneOrMany<all.PatientWorkLocationFieldEnum>;

  /**
   * <https://schema.org/gender>
   **/
  gender: all.OneOrMany<all.PatientGenderFieldEnum>;

  /**
   * <https://schema.org/name>
   **/
  name: all.OneOrMany<string>;

  /**
   * <https://schema.org/mainEntityOfPage>
   **/
  mainEntityOfPage: all.OneOrMany<all.PatientMainEntityOfPageFieldEnum>;

  /**
   * <https://schema.org/url>
   **/
  url: all.OneOrMany<string>;

  /**
   * <https://schema.org/disambiguatingDescription>
   **/
  disambiguatingDescription: all.OneOrMany<string>;

  /**
   * <https://schema.org/identifier>
   **/
  identifier: all.OneOrMany<all.PatientIdentifierFieldEnum>;

  /**
   * <https://schema.org/description>
   **/
  description: all.OneOrMany<all.PatientDescriptionFieldEnum>;

  /**
   * <https://schema.org/subjectOf>
   **/
  subjectOf: all.OneOrMany<all.PatientSubjectOfFieldEnum>;

  /**
   * <https://schema.org/alternateName>
   **/
  alternateName: all.OneOrMany<string>;

  /**
   * <https://schema.org/additionalType>
   **/
  additionalType: all.OneOrMany<all.PatientAdditionalTypeFieldEnum>;

  /**
   * <https://schema.org/potentialAction>
   **/
  potentialAction: all.OneOrMany<all.Action>;

  /**
   * <https://schema.org/sameAs>
   **/
  sameAs: all.OneOrMany<string>;

  /**
   * <https://schema.org/image>
   **/
  image: all.OneOrMany<all.PatientImageFieldEnum>;

  /**
   * <https://schema.org/suggestedAge>
   **/
  suggestedAge: all.OneOrMany<all.QuantitativeValue>;

  /**
   * <https://schema.org/suggestedMinAge>
   **/
  suggestedMinAge: all.OneOrMany<number>;

  /**
   * <https://schema.org/suggestedGender>
   **/
  suggestedGender: all.OneOrMany<all.PatientSuggestedGenderFieldEnum>;

  /**
   * <https://schema.org/requiredMinAge>
   **/
  requiredMinAge: all.OneOrMany<number>;

  /**
   * <https://schema.org/suggestedMeasurement>
   **/
  suggestedMeasurement: all.OneOrMany<all.QuantitativeValue>;

  /**
   * <https://schema.org/requiredGender>
   **/
  requiredGender: all.OneOrMany<string>;

  /**
   * <https://schema.org/requiredMaxAge>
   **/
  requiredMaxAge: all.OneOrMany<number>;

  /**
   * <https://schema.org/suggestedMaxAge>
   **/
  suggestedMaxAge: all.OneOrMany<number>;

  /**
   * <https://schema.org/audienceType>
   **/
  audienceType: all.OneOrMany<string>;

  /**
   * <https://schema.org/geographicArea>
   **/
  geographicArea: all.OneOrMany<all.AdministrativeArea>;
}
